### PR TITLE
Upgrade deps, Add first/last buttons to example, add prop validation

### PR DESCRIPTION
**Pull request by janlazo**
Is the bundle for the example used in some codepen or jsfiddle? It's easy to rebuild after installing the dependencies.

### CODE REVIEW

**Comment by janlazo** [comment]
I'll add prop validation for the remaining directives in case you don't merge it within a few days. I'm going through the typescript code for the smart-table-* packages used by smart-table-core.

Prop validation in Vue is done in runtime and provides warnings,not errors, by default but this is better than nothing.

File: lib/filter.js [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+      smartTable: {
+        type: Object,
+        required: true,
+      },
+      stFilter: {
+        type: String,
+        required: true,
+      },
+      stFilterOperator: {
+        type: String,
```
Comment (janlazo) [inline lib/filter.js]: Can this be a function instead? In https://github.com/smart-table/smart-table-filter/commit/8cfe5ebb962701fd4195621ac876cdd91fe66325#diff-4f0c9b7aee91c4fefb317794d8a1e941R65, allow operator to be a custom function or accept a new option to override `operators[operator]` function.

**Comment by janlazo** [comment]
@lorenzofox3 Any feedback?

**Comment by janlazo** [comment]
@lorenzofox3  I can split up the changes if it's too hard or too much to review in one PR.

**Comment by lorenzofox3** [comment]
Sorry I was in vacations with no internet connection. Also I will have to move the org to gitlab as I live in Cuba, I am under US sanctions and github has started to fuck around...

I'll keep you posted

**Comment by janlazo** [comment]
@lorenzofox3 Any updates?

**Comment by janlazo** [comment]
If you're too busy to review the Vue code or setup the gitlab org, can you delete the generated code for examples and release a new version with updated dependencies? This package works with the latest smart-table-core but adding some usability improvements because of smart-table-core changes would be nice.

**Comment by lorenzofox3** [comment]
Sorry mate for taking so long ... Thanks you for your work and your patience :) 
Note this repo is more a proof of concept, I am in no way a Vuejs expert and do not know what are the current best practices and patterns. Do you ? Would you like to have commit right for this repo ?

**Comment by janlazo** [comment]
Sure, I like to have commit rights to this repo. I am not a VueJS expert as my experience with Vue is only on projects, created by vue-cli and tweaked for SSR. I have some enhancements that I want to port over from my components to the mixins in here in order to improve reactivity and performance. I like to experiment with testing the mixins if you don't mind since I haven't done automated testing on any Vue components I've worked beyond manual testing :sweat_smile: .

**Comment by lorenzofox3** [comment]
Sure, this repo definitely lacks test. You could implement some and make it more robust !

**Pull request merged**